src/Tools/dotnet-dump/Commands/SetThreadCommand.cs

```diff
@@ -4,42 +4,44 @@
 
 using Microsoft.Diagnostics.DebugServices;
 using Microsoft.Diagnostics.Repl;
-using Microsoft.Diagnostics.Runtime;
-using System;
-using System.Collections.Generic;
-using System.CommandLine;
-using System.Linq;
 
 namespace Microsoft.Diagnostics.Tools.Dump
 {
     [Command(Name = "setthread", Help = "Sets or displays the current thread for the SOS commands.")]
     [CommandAlias(Name = "threads")]
     public class SetThreadCommand : CommandBase
     {
-        [Argument(Help = "The thread index to set, otherwise displays the list of threads.")]
-        public int? ThreadIndex { get; set; } = null;
+        [Argument(Help = "The thread index or id to set, otherwise displays the list of threads.")]
```

Comment:
This property is always an index right? I didn't see a case where it was treated as an id.

Reply:
If the "ThreadId" bool is true (--tid/-t option), it will treated as a OS thread id using the thread service's GetThreadInfoFromId.

Reply:
Ah of course you are right - ignore my comment there. I'll blame it on being up too late ; )